python-large-file-upload/doc/azure.md

```diff
@@ -0,0 +1,128 @@
+# Uploading Files to Microsoft Azure via Notecard and Notehub
+
+## Microsoft Azure Setup
+
+1. **Create a Microsoft Azure Account**
+   - If you haven’t done so already, create a Microsoft Azure account at [azure.microsoft.com](https://azure.microsoft.com). Note that you will need to provide a phone number for verification and credit or debit card details for authorization. You won’t be charged for the initial free period unless you switch to pay-as-you-go pricing.
+
+2. **Create a Storage Account**
+   - Once logged into your Azure account, navigate to **Storage accounts**.
+
+     <img src="azure_images/azure_storage_accounts.png" style="border: 1px solid lightgray;">
+
+   - If you don’t have a storage account yet, click **+Create**.
+   - Choose a name for your Storage account. Leave all other options at their default values.
+
+     <img src="azure_images/azure_create_storage_account.png" style="border: 1px solid lightgray;">
+
+   - Click **Review + create** and then **Create**.
+
+3. **Access Your Storage Account**
+   - After the resources are created, click **Go to resource** on the deployment complete screen.
+
+4. **Create a Container**
+   - Navigate to **Data storage** → **Containers**.
+   - Click the **+ Container** button to create a new container.
+   - Name your container (e.g., `bluesfilescontainer`).
+
+5. **Generate SAS Key**
+   - Once the container is created, click the three dots (…) next to your container.
+
+     <img src="azure_images/azure_generate_sas_key_1.png" style="border: 1px solid lightgray;">
+
+   - Click **Generate SAS**.
+   - Select the following permissions:
+     - Read
+     - Add
+     - Create
+     - Write
+     - Delete
+   - The SAS key will have a default expiry time. Optionally, extend this duration for testing to avoid authentication issues.
+   - Click **Generate SAS token and URL** to generate the key.
+   - Copy the **Blob SAS URL** for use in the next steps.
+
+     <img src="azure_images/azure_generate_sas_key_2.png" style="border: 1px solid lightgray;">
+
+
+6. **Test Azure Container Setup**
+   - Copy the generated SAS URL from the previous step.
+   - Add the name you want your file to be saved as in Azure. For example, if you want to save your file as 'datafile', modify the URL by adding a forward slash "/" after the storage container name, followed by the desired file name. For example, before modifying, the URL looks like this:-
+
+     `
+     https://ysazurestorage.blob.core.windows.net/bluesfilescontainer?sp=racw&st=2024-08-01T11:15:56Z&se=2024-08-01T19:15:56Z&spr=https&sv=2022-11-02&sr=c&sig=Kzo3CL%2BkVm74PTdHM5OoxbjsdHyuc1c1KwDnfkcBjJg%3D
```

Comment:
Does this contain any "secrets" or unique IDs?

If so, you may want to replace some of the characters of the unique ID with `xxxxxxxxxxxxxxxxxxxx` or something similar to indicate it's a unique ID, but don't use this specific one.

It's not clear to me, so feel free to ignore this comment if I'm off base.

Reply:
You're right it does, but this specific SAS key was set to expire 24h after it was generated so it is no longer usable.